src/main/kotlin/org/elm/lang/core/types/TypeReplacement.kt

```diff
@@ -44,9 +45,9 @@ class TypeReplacement(
          * `f : (a -> b) -> a -> b`, every time you call the `(a -> b)` parameter within the body of
          * `f`, the ty of `a` and `b` have to be the same.
          */
-        fun freshenVars(ty: Ty): Ty {
-            if (ty.allVars(true).none()) return ty
-            return TypeReplacement(emptyMap(), freshen = true, varsToRemainRigid = null).replace(ty)
+        fun freshenVars(ty: Ty, freeze: Boolean = false): Ty {
+            if (!tyWouldChange(ty, freeze)) return ty
+            return TypeReplacement(emptyMap(), freshen = true, varsToRemainRigid = null, freeze = false).replace(ty)
```

Comment:
This is probably ok, but just to double-check, is it ok that the `freeze` param is always `false` here?

Reply:
Nope, that's a copy-paste error. Good catch!